lib/Myriad/RPC/Implementation/Memory.pm

```diff
@@ -67,11 +67,12 @@ async method start () {
                     $messages->{$id}->{transport_id} = $id;
                     $message = Myriad::RPC::Message::from_hash($messages->{$id}->%*);
                     $rpc->{sink}->emit($message);
-                } catch ($e isa Myriad::Exception::RPC::BadEncoding) {
-                    $log->warnf('Recived a dead message that we cannot parse, going to drop it.');
-                    $log->tracef("message was: %s", $messages->{$id});
-                    await $self->drop($rpc->{stream}, $id);
                 } catch ($e) {
+                    if (blessed $e && $e->isa('Myriad::Exception::RPC::BadEncoding')) {
+                        $log->warnf('Recived a dead message that we cannot parse, going to drop it.');
+                        $log->tracef("message was: %s", $messages->{$id});
+                        await $self->drop($rpc->{stream}, $id);
```

Comment:
This changes the logic? I think it'd need to be `if ... else` rather than falling through to the default case.